numtrip/game.py

```diff
@@ -31,11 +33,26 @@ def rows (self):
     def columns():
         print(Numtrip.y[Numtrip.coordinateY], end="")
         Numtrip.coordinateY = Numtrip.coordinateY + 1
+    def analyse(self):
+        coordinate = Numtrip.wanted.strip()
+        """
```

Comment:
Cool, Sie verwenden bereits Klassen. Kennen Sie den Unterschied zwischen Instanzvariablen und Klassenvariablen?

Wozu braucht man das `self`? Warum plötzlich `Numtrip.wanted`?

--> Wenn Sie den Klassen-basierten Ansatz wählen, so müssen Sie die verwendeten Konzepte beim Fachgespräch erklären können.

Reply:
Klassenvariablen können alles enthalten (soviel ich weiss) und zwar: Funktionen, Variablen, Listen(Arrays) und Loops.
Die Klassenvariabeln dienen als eine Art Liste, und/oder Vorlagen für die Instanzvariabeln. Wenn man nichts in den Instanzvariabeln definiert, so wird sie eine Kopie von der Klassenvariable, wird aber etwas hingeschrieben so ändern sich die gewollten Werte, doch die Klassenvariable bleibt unverändert.

Das habe ich eigentlich neugeschrieben, jedoch will GitHub die neuste Version nicht hochladen (es ist nicht das erste Mal und auch nicht das einzige Dokument)...
`self `nutzt man hier für das Gleiche wie `Numtrip`, das greift auf die Werte , welche in der Klassenvariable `Numtrip `definiert wurden in diesem Fall `wanted`, aber `Numtrip.wanted` (`self.wanted`) ist eine Instanzvariabel.